lib/cnet/punt/punt_kernel.c

```diff
@@ -37,40 +38,47 @@
 #include <cnet_eth.h>
 #include <net/cne_udp.h>
 #include <hexdump.h>
+#include <cnet_tcp.h>
 
 #include <cnet_node_names.h>
 #include "punt_kernel_priv.h"
-#include "tun_alloc.h"
 
 #define PREFETCH_CNT 6
 
 static __cne_always_inline void
 punt_kernel_process_mbuf(struct cne_node *node, pktmbuf_t **mbufs, uint16_t cnt)
 {
     punt_kernel_node_ctx_t *ctx = (punt_kernel_node_ctx_t *)node->ctx;
-    struct iovec v[cnt];
-    int fd;
 
-    if ((fd = tun_get_fd(ctx->tinfo)) >= 0) {
+    if (ctx->sock >= 0) {
+        struct cne_ipv4_hdr *ip4;
+        struct sockaddr_in sin = {0};
+        size_t len;
+        char *buf;
+
         for (int i = 0; i < cnt; i++) {
-            v[i].iov_base = pktmbuf_mtod(mbufs[i], void *);
-            v[i].iov_len  = pktmbuf_data_len(mbufs[i]);
+            ip4 = pktmbuf_mtod(mbufs[i], struct cne_ipv4_hdr *);
+            len = pktmbuf_data_len(mbufs[i]);
+            buf = (char *)ip4;
+
+            sin.sin_family      = AF_INET;
+            sin.sin_port        = 0;
+            sin.sin_addr.s_addr = ip4->dst_addr;
+
+            if (sendto(ctx->sock, buf, len, 0, (struct sockaddr *)&sin, sizeof(sin)) < 0)
+                CNE_WARN("Unable to send packets: %s\n", strerror(errno));
```

Comment:
This log message could be overkill under high load. Ok for now, but eventually we'll want a stat increment or something rather than a log.

Reply:
I agree, it needs to be a stats counter.